internal/runtime/hcsv2/standalone_container.go

```diff
@@ -0,0 +1,139 @@
+package hcsv2
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/opengcs/internal/network"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func getStandaloneRootDir(id string) string {
+	return filepath.Join("/tmp/gcs/s", id)
+	return filepath.Join("/tmp/gcs/s", id)
+}
+
+func getStandaloneHostnamePath(id string) string {
+	return filepath.Join(getStandaloneRootDir(id), "hostname")
+}
+
+func getStandaloneHostsPath(id string) string {
+	return filepath.Join(getStandaloneRootDir(id), "hosts")
+}
+
+func getStandaloneResolvPath(id string) string {
+	return filepath.Join(getStandaloneRootDir(id), "resolv.conf")
+}
+
+func setupStandaloneContainerSpec(ctx context.Context, id string, spec *oci.Spec) (err error) {
+	// TODO: JTERRY75 use ctx for log
+	operation := "setupStandaloneContainerSpec"
+	start := time.Now()
+	defer func() {
+		end := time.Now()
+		fields := logrus.Fields{
+			"cid":      id,
+			"endTime":  end,
+			"duration": end.Sub(start),
+		}
+		if err != nil {
+			fields[logrus.ErrorKey] = err
+			logrus.WithFields(fields).Error(operation)
+		} else {
+			logrus.WithFields(fields).Info(operation)
+		}
+	}()
+
+	// Generate the standalone root dir
+	rootDir := getStandaloneRootDir(id)
+	if err := os.MkdirAll(rootDir, 0755); err != nil {
+		return errors.Wrapf(err, "failed to create container root directory %q", rootDir)
+	}
+
+	// Write the hostname
+	if !isInMounts("/etc/hostname", spec.Mounts) {
```

Comment:
There are some differences between what we do here and for a sandbox container that I don't understand:

- For sandbox we don't check if the file is in mounts before generating it. Should we do that?
- For standalone, it seems we add another mount for /etc/hostname. Does this mean we'll have two mount entries for it?
- Should we also be adding a mount entry for the sandbox case?

Reply:
- No standalone containers have different semantics. We are forced with the sandbox to create the file every time. Only the workload container is allowed to override. The standalone container is more like the "sandbox + workload" all at once.
- We only have one mount for each of `/etc/hosts`, `/etc/hostname` and `/etc/resolv.conf`
- Sandbox actually has no workload so it needs no mounts. It just for "accounting" to make sure all workloads share these files that they are generated as part of the "sandbox" activation

Reply:
I think I misread `!isInMounts(...)` and missed the `!`. Thus my confusion. :)